File: backend/app/api/v1/max_ui.py
from __future__ import annotations

from fastapi import APIRouter

from app.max_ui.catalog import list_product_types

router = APIRouter()


@router.get("/max-ui/product-types", tags=["max_ui"])
def get_max_ui_product_types():
    return {"productTypes": list_product_types()}

File: frontend/lib/api/max-ui.ts
import { apiGet } from "@/lib/api/client";

export type MaxUiProductTypesResponse = {
  productTypes: string[];
};

export async function getMaxUiProductTypes(): Promise<MaxUiProductTypesResponse> {
  return apiGet<MaxUiProductTypesResponse>("/max-ui/product-types");
}

